src/debugger/appWorker.ts

```diff
@@ -217,8 +223,23 @@ export class MultipleLifetimesAppWorker {
         this.socketToApp.on("open", () => {
             this.onSocketOpened();
         });
-        this.socketToApp.on("close", () =>
-            this.onSocketClose());
+        this.socketToApp.on("close",
+            () => {
+                this.executionLimiter.execute("onSocketClose.msg", /*limitInSeconds*/ 10, () => {
+                    /*
+                     * It is not the best idea to compare with the message, but this is the only thing React Native gives that is unique when
+                     * it closes the socket because it already has a connection to a debugger.
+                     * https://github.com/facebook/react-native/blob/588f01e9982775f0699c7bfd56623d4ed3949810/local-cli/server/util/webSocketProxy.js#L38
+                     */
+                    if (this.socketToApp._closeMessage === "Another debugger is already connected") {
```

Comment:
If another debugger is already connected, should we still attempt to reconnect? Or are you assuming that by rejecting the promise the process will terminate and so it doesn't matter?